tests/ui-toml/toml_unknown_key/conf_unknown_key.rs

```diff
@@ -1,3 +1,4 @@
+//@no-rustfix
```

Comment:
The setup for this test is a bit confusing. Rustfix wants to think that this file is clippy.toml. If I remove this line, it errors with
```
error: failed to apply suggestions for tests/ui-toml/toml_unknown_key/conf_unknown_key.rs with rustfix
Invalid range 133...165 given, original data is only 78 byte long
Add //@no-rustfix to the test file to ignore rustfix suggestions
```

Reply:
This happens because this whole file is 78 bytes long. Weird, but we should probably fix it before approval

Reply:
I'm not sure *why* it wants to make any changes to `conf_unknown_key.rs` in the first place though... Like, the span applies to the `config.toml` file? Gonna have to look into this

Reply:
Looking into this, it seems like uitest always passes the contents of the rust file being tested to rustfix, regardless of where the actual suggestion comes from.

Here it asks rustfix to parse out the errors and suggestions:
https://github.com/oli-obk/ui_test/blob/6b567fdb2efed6c967e5b739b20a751a1968a935/src/lib.rs#L853-L856

... and then it always uses the same input `path`, even though it's totally possible for the actual path of the diagnostic to be in a different file:
https://github.com/oli-obk/ui_test/blob/6b567fdb2efed6c967e5b739b20a751a1968a935/src/lib.rs#L877-L878

Do you think it's fine if I open an issue on the repo and just keep the no-rustfix annotation for now?

Reply:
Put a comment with a `// TODO: <PR LINK (AND SOME DESCRIPTION OF WHY THE ATTRIBUTE IS THERE)>`, let's hope that either Oli or any other contributor has some time to fix it.

I'll add it to my backlog and check in a month or so.